Modix.Services/Core/ChannelService.cs

```diff
@@ -29,33 +41,62 @@ public class ChannelService : IChannelService
         /// Constructs a new <see cref="ChannelService"/> with the given injected dependencies.
         /// </summary>
         public ChannelService(
-            IGuildChannelRepository guildChannelRepository)
+            IGuildChannelRepository guildChannelRepository,
+            IGuildService guildService,
+            IMemoryCache cache)
         {
             _guildChannelRepository = guildChannelRepository;
+            _guildService = guildService;
+            _cache = cache;
         }
 
         /// <inheritdoc />
         public async Task TrackChannelAsync(IGuildChannel channel, CancellationToken cancellationToken = default)
         {
-            using (var transaction = await _guildChannelRepository.BeginCreateTransactionAsync(cancellationToken))
+            using var transaction = await _guildChannelRepository.BeginCreateTransactionAsync(cancellationToken);
+
+            if (channel is ITextChannel textChannel)
             {
-                if (!await _guildChannelRepository.TryUpdateAsync(channel.Id, data =>
+                var key = GetKey(channel.GuildId, channel.Id);
+                _cache.Set(key, textChannel, TimeSpan.FromDays(7));
+            }
+
+            if (!await _guildChannelRepository.TryUpdateAsync(channel.Id, data =>
                 {
                     data.Name = channel.Name;
                 }, cancellationToken))
+            {
+                await _guildChannelRepository.CreateAsync(new GuildChannelCreationData()
                 {
-                    await _guildChannelRepository.CreateAsync(new GuildChannelCreationData()
-                    {
-                        ChannelId = channel.Id,
-                        GuildId = channel.GuildId,
-                        Name = channel.Name
-                    }, cancellationToken);
-                }
-
-                transaction.Commit();
+                    ChannelId = channel.Id,
+                    GuildId = channel.GuildId,
+                    Name = channel.Name
+                }, cancellationToken);
+            }
+
+            transaction.Commit();
+        }
+
+        public async ValueTask<ITextChannel> GetChannelAsync(ulong guildId, ulong channelId)
```

Comment:
A) The Discord.NET Socket client already caches ALL guild data in memory, except for messages.
B) I'm very much not comfortable caching objects for which we do not control the lifetime.

Reply:
That's fair, I wasn't sure how much D.Net cached

Reply:
Have a look at the API of the `Socket` classes, you can usually tell. `SocketGuild`, for example, has a property` public IReadOnlyCollection<SocketGuildChannel> Channels { get; }` and a method `public SocketGuildChannel GetChannel(ulong id);`